src/main/java/org/matsim/episim/InfectionEventHandler.java

```diff
@@ -488,12 +494,67 @@ public void init(Map<DayOfWeek, List<Event>> events) {
 		personMap.values().forEach(p -> p.getSpentTime().clear());
 		personMap.values().forEach(EpisimPerson::initParticipation);
 
+		balanceContainersByLoad(estimatedLoad);
+		
 		createTrajectoryHandlers();
-
+		
 		init = true;
 	}
 
-
+	/**
+	 * Distribute the containers to the different ReplayEventTasks, by setting
+	 * the taskId attribute of the containers to values between 0 and episimConfig.getThreds() - 1,
+     * so that the sum of numUsers * maxGroupSize has an even distribution	
+	 */
+	private void balanceContainersByLoad(List<Tuple<EpisimContainer<?>, Double>> estimatedLoad) {
+		// We need the containers sorted by the load, with the highest load first
+		Collections.sort(estimatedLoad, new Comparator<Tuple<EpisimContainer<?>, Double>>() {
+			@Override
+			public int compare(Tuple<EpisimContainer<?>, Double> t1, Tuple<EpisimContainer<?>, Double> t2) {
+				if (Math.abs(t1.getSecond() - t2.getSecond()) < 1.0e-9)
+					return 0;
+				else if (t1.getSecond() < t2.getSecond())
+					return 1;
+				return -1;
+			}
+		});
+	
+		final int numThreads = episimConfig.getThreads();
+		// the overall load of the containers assigned to the thread/taskId
+		final Double[] loadPerThread = new Double[numThreads];
+		for (int i = 0; i < numThreads; i++)
+			loadPerThread[i] = 0.0;
+		
+		for(Tuple<EpisimContainer<?>, Double> tuple : estimatedLoad) {
+			// search for the thread/taskId with the minimal load
+			int useThread = 0;
+			Double minLoad = loadPerThread[0];
+			for (int i = 1; i < numThreads; i++) {
+				if (loadPerThread[i] < minLoad) {
+					useThread = i;
+					minLoad = loadPerThread[i];
+				}
+			}
+			// add the load to this thread and set the taskId for the container
+			loadPerThread[useThread] += tuple.getSecond();
+			tuple.getFirst().setTaskId(useThread);
+			if (useThread == 0)
+				log.info("In task 0: {}", tuple.getFirst().getContainerId());
```

Comment:
There is a lot of noise now in the log. Probably this should be `trace` level?

Reply:
Sorry, this logging sneaked into the commit, i removed it now.